code/datums/ai/snake/snake_behaviors.dm

```diff
@@ -0,0 +1,101 @@
+
+
+///snakes will retreat after biting someone as much as they deserve, since they're not trying to kill you.
+/datum/ai_behavior/snake_flee
+
+/datum/ai_behavior/snake_flee/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	var/mob/living/living_pawn = controller.pawn
+	var/mob/living/target = controller.blackboard[BB_SNAKE_CURRENT_RETREAT_TARGET]
+
+	if(!target)
+		finish_action(controller, TRUE)
+	var/nearby = FALSE
+	for(var/mob/living/L in view(living_pawn, 7))
+		if(L == target)
+			nearby = TRUE
+
+	if(nearby)
+		walk_away(living_pawn, target, 7, 8)// we're gonna run from someone we've bitten to satisfaction
+	else
+		finish_action(controller, TRUE)
+
+/datum/ai_behavior/snake_flee/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	var/mob/living/living_pawn = controller.pawn
+	walk(living_pawn, 0)
+	controller.blackboard[BB_SNAKE_CURRENT_RETREAT_TARGET] = null
+
+/datum/ai_behavior/snake_attack_mob
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM //performs to increase frustration
+
+/datum/ai_behavior/snake_attack_mob/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	var/mob/living/target = controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET]
```

Comment:
monki

Reply:
theres a few other occurrences of what looks like monkey defines when it should be snake defines, but i may just be misunderstanding something

Reply:
FRIDGE! Thanks for catching that